Source/com/Communicator.h

```diff
@@ -421,10 +425,35 @@ namespace RPC {
             return (result);
         }
 
+    private:
+        const std::vector<string> DynamicLoaderPaths() const;
+        void PopulateLDLibraryPaths(const string& oldLDLibraryPaths, string& newLDLibraryPaths) const {
+            // Read currently added LD_LIBRARY_PATH to prefix with _systemRootPath
+            if (oldLDLibraryPaths.empty() != true) {
+                size_t start = 0;
+                size_t end = oldLDLibraryPaths.find(':');
+                do {
+                    newLDLibraryPaths += _systemRootPath;
+                    newLDLibraryPaths += oldLDLibraryPaths.substr(start,
+                                        ((end != string::npos) ? (end - start + 1) : end));
+                    start = end;
+                    if (end != string::npos) {
+                        end = oldLDLibraryPaths.find(':', start + 1);
+                    }
+                } while (start != string::npos);
+            }
+
+            std::vector<string> loaderPaths = DynamicLoaderPaths();
```

Comment:
Just accept a const std::vector<string>& here to avoid a copy..

Reply:
done